x-pack/plugins/case/server/routes/api/__fixtures__/authc_mock.ts

```diff
@@ -3,33 +3,22 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import { Authentication } from '../../../../../security/server';
+import { AuthenticatedUser } from '../../../../../security/server';
+import { securityMock } from '../../../../../security/server/mocks';
 
-const getCurrentUser = jest.fn().mockReturnValue({
-  username: 'awesome',
-  full_name: 'Awesome D00d',
-});
-const getCurrentUserThrow = jest.fn().mockImplementation(() => {
-  throw new Error('Bad User - the user is not authenticated');
-});
+function createAuthenticationMock({
+  currentUser,
+}: { currentUser?: AuthenticatedUser | null } = {}) {
+  const { authc } = securityMock.createSetup();
+  authc.getCurrentUser.mockReturnValue(
+    currentUser !== undefined
+      ? currentUser
+      : ({ username: 'awesome', full_name: 'Awesome D00d' } as AuthenticatedUser)
+  );
+  return authc;
+}
 
 export const authenticationMock = {
```

Comment:
__note:__ consumers are supposed to use mocks provided by the plugin itself and to not redefine it to be as much in sync with real types as possible.